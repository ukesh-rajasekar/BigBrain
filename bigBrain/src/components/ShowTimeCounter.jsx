import React,{useState, useEffect} from 'react'

export default function ShowTimeCounter({startTime, timeLimit, handleTimeout}) {
    console.log(startTime, timeLimit);
    const [count, setCounter] = useState(Number(timeLimit))
    useEffect(() => {
        const counter = setInterval(() => {
            const now = new Date()
            const time_now = now.getTime()
            const start = new Date(startTime)
            const time_start = start.getTime() 
            const time = Math.floor((time_now-time_start)/1000);
            if((timeLimit - time )< 0){
                handleTimeout()
                clearInterval(counter)
            }else{
                setCounter(time)

            }
            }, 1000);
        return () => {
            clearInterval(counter)
        }
    }, [count])
   
    return (
        <div>
            {timeLimit - count}
        </div>
    )
}
