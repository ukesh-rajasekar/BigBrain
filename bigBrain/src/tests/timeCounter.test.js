import React from 'react'
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ShowTimeCounter from '../components/ShowTimeCounter'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



describe('Render showTimer', () => {
  it('render with minimal props', () => {  
    const TimerComponent = renderer.create(<ShowTimeCounter />).toJSON();
    expect(TimerComponent).toMatchSnapshot();
  });

  it('uses startTime,timeLimit to calculate timeout', () => {  
    const startTime = "2021-04-22T03:24:31.196Z"
    const timeLimit = "60"
    const TimerComponent = renderer.create(<ShowTimeCounter startTime = {startTime} timeLimit={timeLimit}/>).toJSON();
    expect(TimerComponent).toMatchSnapshot();
  });

  
  });
