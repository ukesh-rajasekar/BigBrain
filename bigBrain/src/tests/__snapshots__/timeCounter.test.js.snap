// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render showTimer render with minimal props 1`] = `
<div>
  Time Left: 
  NaN
</div>
`;

exports[`Render showTimer uses startTime,timeLimit to calculate timeout 1`] = `
<div>
  Time Left: 
  0
</div>
`;
